Changes get_url.sha256sum to get_url.checksum

histamineblkr opened the pull request:
The get_url.sha256sum has been deprecated in favor of the get_url.checksum parameter. Corresponding Ansible documentation: https://docs.ansible.com/ansible/2.7/modules/get_url_module.html

histamineblkr committed:
The get_url.sha256sum has been deprecated in favor of the get_url.checksum parameter.

Corresponding Ansible documentation: https://docs.ansible.com/ansible/2.7/modules/get_url_module.html

histamineblkr committed:
Implements the ansible yum module install of the datadog-agent package.

The ansible issue: https://github.com/ansible/ansible/issues/20608 has been closed and the ansible yum module works as expected for the datadog-agent package install.

histamineblkr commented:
I accidentally pushed the yum module change when I was cleaning up my old work laptop to move my repositories over to the new one. I was going to make a separate PR for the yum module since the issue has been closed: https://github.com/ansible/ansible/issues/20608
I have tested this in our local repo and the yum module works as expected. I can close this branch and do the checksum and yum in separate branches or leave them together in this one. Please let me know which you prefer.

hush-hush commented on tasks/pkg-redhat.yml:
```diff
@@ -3,7 +3,7 @@
     url: "{{ datadog_yum_gpgkey_new }}"
     dest: /tmp/DATADOG_RPM_KEY_E09422B3.public
-    sha256sum: 694a2ffecff85326cc08e5f1a619937999a5913171e42f166e13ec802c812085
+    checksum: "sha256:694a2ffecff85326cc08e5f1a619937999a5913171e42f166e13ec802c812085"
```
Could you do the same in `tasks/pkg-suse.yml` ?

histamineblkr replied:
Yes I will do it today.

histamineblkr replied:
Change made. Do you have a walk through how to use your tests for the this role? I'm not sure how to test the Suse change above.

hush-hush reviewed (changes requested):
Thanks for this PR @histamineblkr !

Left a quick comment but it looks good otherwise.

albertvaka commented:
Hi @histamineblkr, can you make the fix pointed out? So this can get merged.

histamineblkr commented:
Yes I will today, I apologize for the slow response.

Commit:
Changes checksum method.

The get_url.sha256sum has been deprecated in favor of the get_url.checksum parameter.
Corresponding Ansible documentation: https://docs.ansible.com/ansible/2.7/modules/get_url_module.html

histamineblkr commented:
@hush-hush changes were made a bit ago to the pkg-suse.yml file so this commit should be good to go now.

dabcoder commented:
@histamineblkr Sorry for the delay. Looks good, thanks.

> I can close this branch and do the checksum and yum in separate branches or leave them together in this one. Please let me know which you prefer.

Actually the yum install issue is being taken care of in another PR, currently in review: https://github.com/DataDog/ansible-datadog/pull/144. Mind removing that part then (but thanks for raising this point regardless)?

dabcoder commented:
Took care of it in the linked PR, thanks for initiating this @histamineblkr
